readthedocs/core/middleware.py

```diff
@@ -57,21 +54,21 @@ def process_request(self, request):
                     raise Http404(_('Project not found'))
                 request.subdomain = True
                 request.slug = subdomain
-                request.urlconf = SUBDOMAIN_URLCONF
+                request.urlconf = settings.SUBDOMAIN_URLCONF
                 return None
 
         # Serve CNAMEs
         if (
-            public_domain not in host and production_domain not in host and
-            'localhost' not in host and 'testserver' not in host
+            public_domain not in host and settings.PRODUCTION_DOMAIN not in host
+            and 'localhost' not in host and 'testserver' not in host
```

Comment:
we don't start a line with an operator

Reply:
Also, maybe this can be written like

```python
if  all(test_host not in host for test_host in ('localhost', 'testserver', public_domain, settings.PRODUCTION_DOMAIN)):
```

(not sure if that is more readable)